test/test_nn.py

```diff
@@ -16031,6 +16031,12 @@ def embedding_bag_check(indices, weights, mode, sparse, padding_idx):
                     rtol = None
                 self.assertEqual(grad, grad_check, msg=msg, atol=atol, rtol=rtol)
 
+    def _slow_masked_softmax(self, input, mask):
```

Comment:
This gives you awful numerical stability. Why not something like
```
def masked_softmax(input, dim, _stacklevel=3, dtype=None, mask=None):
    return torch.nn.functional.softmax(input.masked_fill(~mask, float('-inf')),                  
                                       dim, _stacklevel, dtype)
```

Reply:
because masked_softmax_cuda will do just that for some inputs: https://github.com/pytorch/pytorch/blob/35d4a805ebc3b6eca1bafb2d332dffa8d0c1fc54/aten/src/ATen/native/cuda/SoftMax.cu#L976

Reply:
also, this is not new code; it is a move of a function that was already present. (see line 16162 below)